### ISSUE TITLE
AttributeError: 'dict' object has no attribute 'name'

### ISSUE THREAD

**Issue by nbro** [tensorflow/tensorflow#38988]
**System information**
- Have I written custom code: No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS X Catalina
- TensorFlow installed from: binary
- TensorFlow version: ` 2.2.0.dev20200427` 
- Python version: 3.7.5

**Describe the current behavior**

I get the error

> AttributeError: 'dict' object has no attribute 'name'

when I try to plot a model, with `tf.keras.utils.plot_model`, where I specify the loss to the `compile` method as a dictionary. According to [the documentation](https://www.tensorflow.org/api_docs/python/tf/keras/Model?version=nightly#compile), this should be possible and this was actually possible in TF 2.1 (i.e. until recently I was using the exact same code with TF 2.1 and no error was thrown).

**Describe the expected behavior**

No error.

**Standalone code to reproduce the issue**

```
import tensorflow as tf


def get_model(input_shape, num_classes=10):
    model_input = tf.keras.layers.Input(shape=input_shape)
    x = tf.keras.layers.Conv2D(6, 3)(model_input)
    x = tf.keras.layers.Flatten()(x)
    model_output = tf.keras.layers.Dense(num_classes, name="my_output_layer")(x)
    model = tf.keras.Model(model_input, model_output)
    model.summary()
    return model


def train():
    model = get_model((28, 28, 1))

    model.compile(loss={"my_output_layer": "categorical_crossentropy"})

    # IF YOU COMMENT THIS, NO ERROR OCCURS!!!
    tf.keras.utils.plot_model(model, to_file="model.png", show_shapes=True)


if __name__ == '__main__':
    train()
```

I installed `pydot` with `pip install pydot`. The version that was installed is `1.4.1`.

Clearly, the error is due to the fact that `plot_model` thinks that the loss is also a layer.

I need to use TF 2.2 because I am using another library that requires the nightly version of TF.

**Comment by Saduf2019** [comment]
@nbro
I ran the code shared above and do not face any error on nightly, please find the [gist here.](https://colab.sandbox.google.com/gist/Saduf2019/cdf44239edc5dd83d9574f7b2ad6f5ef/untitled162.ipynb)

**Comment by nbro** [comment]
@Saduf2019 If I run your notebook, I get the error I am reporting above. Please, run it again!

**Comment by jscant** [comment]
I get this error in tf 2.1, but only if I use
`
weights = Input(shape=(1,), dtype=tf.float32, name='l2_weight')
model = tf.keras.Model(
    inputs=tuple([input, weights, y_true]),
    outputs=tuple([y_pred]))
model.add_loss(
    weighted_squared_error(y_true, y_pred, weights), name='weighted_loss')
model.compile(optimizer=opt(learning_rate=lr))
`

Using this code works fine, but it's no use because I want to be able to use a custom loss function with weights as an input:
`
model = tf.keras.Model(
    inputs=input, outputs=y_pred)
model.compile(optimizer=opt(lr=lr), loss='binary_crossentropy')
`

**Comment by Saduf2019** [comment]
@nbro
Yes i confirm on rerunning the gist i am able to see the error.

**Comment by jscant** [comment]
When this happens, tensorflow appears to have appended an empty dictionary to `model._layers`. A workaround for the moment is, once the model has been compiled, to use:

```
model._layers = [layer for layer in model._layers if not isinstance(layer, dict)]
```


> **System information**
> 
> * Have I written custom code: No
> * OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS X Catalina
> * TensorFlow installed from: binary
> * TensorFlow version: ` 2.2.0.dev20200427`
> * Python version: 3.7.5
> 
> **Describe the current behavior**
> 
> I get the error
> 
> > AttributeError: 'dict' object has no attribute 'name'
> 
> when I try to plot a model, with `tf.keras.utils.plot_model`, where I specify the loss to the `compile` method as a dictionary. According to [the documentation](https://www.tensorflow.org/api_docs/python/tf/keras/Model?version=nightly#compile), this should be possible and this was actually possible in TF 2.1 (i.e. until recently I was using the exact same code with TF 2.1 and no error was thrown).
> 
> **Describe the expected behavior**
> 
> No error.
> 
> **Standalone code to reproduce the issue**
> 
> ```
> import tensorflow as tf
> 
> 
> def get_model(input_shape, num_classes=10):
>     model_input = tf.keras.layers.Input(shape=input_shape)
>     x = tf.keras.layers.Conv2D(6, 3)(model_input)
>     x = tf.keras.layers.Flatten()(x)
>     model_output = tf.keras.layers.Dense(num_classes, name="my_output_layer")(x)
>     model = tf.keras.Model(model_input, model_output)
>     model.summary()
>     return model
> 
> 
> def train():
>     model = get_model((28, 28, 1))
> 
>     model.compile(loss={"my_output_layer": "categorical_crossentropy"})
> 
>     # IF YOU COMMENT THIS, NO ERROR OCCURS!!!
>     tf.keras.utils.plot_model(model, to_file="model.png", show_shapes=True)
> 
> 
> if __name__ == '__main__':
>     train()
> ```
> 
> I installed `pydot` with `pip install pydot`. The version that was installed is `1.4.1`.
> 
> Clearly, the error is due to the fact that `plot_model` thinks that the loss is also a layer.
> 
> I need to use TF 2.2 because I am using another library that requires the nightly version of TF.

When this happens, tensorflow appears to have appended an empty dictionary to `model._layers`. A workaround for the moment is, once the model has been compiled, to use: `model._layers = [layer for layer in model._layers if not isinstance(layer, dict)]`

Your example would therefore read:
```
import tensorflow as tf


def get_model(input_shape, num_classes=10):
    model_input = tf.keras.layers.Input(shape=input_shape)
    x = tf.keras.layers.Conv2D(6, 3)(model_input)
    x = tf.keras.layers.Flatten()(x)
    model_output = tf.keras.layers.Dense(num_classes, name="my_output_layer")(x)
    model = tf.keras.Model(model_input, model_output)
    model.summary()
    return model


def train():
    model = get_model((28, 28, 1))

    model.compile(loss={"my_output_layer": "categorical_crossentropy"})
    model._layers = 
    # IF YOU COMMENT THIS, NO ERROR OCCURS!!!
    tf.keras.utils.plot_model(model, to_file="model.png", show_shapes=True)


if __name__ == '__main__':
    train()
```

**Comment by nixingyang** [comment]
For some reasons, the last layer in my model._layers is a ```ListWrapper```, and I am using multiple loss functions.
Thus, I am using the following snippet as a workaround:
```python
from tensorflow.keras.layers import Layer
model._layers = [
    layer for layer in model._layers if isinstance(layer, Layer)
]
plot_model(model)
```

**Comment by ymodak** [comment]
This is fixed with tf-nightly 2.4.0-dev20200828.
See [gist](https://colab.research.google.com/gist/ymodak/c20e2929379074ffc1f3fa80ae258b09/github_38988.ipynb) for your reference. Thanks!
